src/client/components/localisation/skybox/view_model.ts

```diff
@@ -0,0 +1,80 @@
+import { createTransformer } from 'mobx'
+import { computed } from 'mobx'
+import { Mesh } from 'three'
+import { SphereBufferGeometry } from 'three'
+import { MeshBasicMaterial } from 'three'
+import { PlaneBufferGeometry } from 'three'
+import { SkyboxModel } from './model'
+
+export class SkyboxViewModel {
+  public constructor(private model: SkyboxModel) {
+  }
+
+  public static of = createTransformer((model: SkyboxModel): SkyboxViewModel => {
+    return new SkyboxViewModel(model)
+  })
+
+  @computed
+  public get skybox(): SkyBoxProps {
```

Comment:
It would be ideal to separate this function into multiple `@computed` properties which represented each `Mesh`/`Material`/`Geometry`. Not only just to keep functions small, but also for performance reasons. Mobx will cache each `@computed` property if no model changes occur that would require its re-computation. This is why I keep the geometries in the own `@computed` property since they are expensive to create and are otherwise completely static.

e.g. if the `luminance ` uniform were to change, we don't need to regenerate the ground, since it doesn't depend on the `luminance` uniform property. With it all being in one function like this, it would need to recompute all of it, when really only the sky changes.

Maybe use https://github.com/NUbots/NUsight2/blob/8715706c7e5b5034c04058bbb2eef027d9c2a531/src/client/components/localisation/field/view_model.ts#L21 as a good reference.